it/v1/endpoints/CreateForeignPropertyControllerISpec.scala

```diff
@@ -0,0 +1,645 @@
+        response.status shouldBe Status.BAD_REQUEST
+        response.json shouldBe Json.toJson(BadRequestError)
+      }
+    }
+
+    "return error according to spec" when {
+      "validation error" when {
+        def validationErrorTest(requestNino: String, requestBusinessId: String, requestBody: JsValue, expectedStatus: Int, expectedBody: MtdError): Unit = {
+          s"validation fails with ${expectedBody.code} error" in new CreateTest {
+
+            override val nino: String = requestNino
+            override val businessId: String = requestBusinessId
+
+            override def setupStubs(): StubMapping = {
+              AuditStub.audit()
+              AuthStub.authorised()
+              MtdIdLookupStub.ninoFound(nino)
+            }
+
+            val response: WSResponse = await(request().post(requestBody))
+            response.status shouldBe expectedStatus
+            response.json shouldBe Json.toJson(expectedBody)
+          }
+        }
+
+        val input = Seq(
+          ("AA1123A", "XAIS12345678910", requestJson, Status.BAD_REQUEST, NinoFormatError),
+          ("AA123456A", "XA***IS1", requestJson, Status.BAD_REQUEST, BusinessIdFormatError),
+          ("AA123456A", "XAIS12345678910", invalidToDateRequestJson, Status.BAD_REQUEST, ToDateFormatError),
+          ("AA123456A", "XAIS12345678910", invalidFromDateRequestJson, Status.BAD_REQUEST, FromDateFormatError),
+          ("AA123456A", "XAIS12345678910", invalidCountryCodeRequestJson, Status.BAD_REQUEST, allInvalidCountryCodeRequestError),
+          ("AA123456A", "XAIS12345678910", invalidValueRequestJson, Status.BAD_REQUEST, allInvalidValueRequestError),
+          ("AA123456A", "XAIS12345678910", bothExpensesSuppliedRequestJson, Status.BAD_REQUEST, RuleBothExpensesSuppliedRequestError),
+          ("AA123456A", "XAIS12345678910", toDateBeforeFromDateRequestJson, Status.BAD_REQUEST, RuleToDateBeforeFromDateError),
+          ("AA123456A", "XAIS12345678910", ruleCountryCodeErrorRequestJson, Status.BAD_REQUEST, allRuleCountryCodeRequestError)
+        )
+
+        input.foreach(args => (validationErrorTest _).tupled(args))
+      }
+      "des service error" when {
+        def serviceErrorTest(desStatus: Int, desCode: String, expectedStatus: Int, expectedBody: MtdError): Unit = {
+          s"des returns an $desCode error and status $desStatus" in new CreateTest {
+
+            override def setupStubs(): StubMapping = {
+              AuditStub.audit()
+              AuthStub.authorised()
+              MtdIdLookupStub.ninoFound(nino)
+              DesStub.onError(DesStub.POST, desUri, desStatus, errorBody(desCode))
+            }
+
+            val response: WSResponse = await(request().post(requestJson))
+            response.status shouldBe expectedStatus
+            response.json shouldBe Json.toJson(expectedBody)
+          }
+        }
+
+        val input = Seq(
```

Comment:
You're not testing a lot of the DES responses here - you need to include tests for DES returning:
- INVALID_TAXABLE_ENTITY_ID (400)
- FORMAT_BUSINESS_ID (400)
- RULE_OVERLAPPING_PERIOD (422)
- RULE_MISALIGNED_PERIOD (422)
- RULE_NOT_CONTIGUOUS_PERIOD (422)